RFC-0005-multi-cluster-resource-management.md

```diff
@@ -0,0 +1,64 @@
+* https://github.com/prestodb/presto/pull/12176
+* https://github.com/trinodb/trino/issues/391
+* https://github.com/prestodb/presto/issues/10174
+* https://github.com/prestodb/presto/issues/13198
+
+## Background
+
+Currently, each Presto cluster owns its own resource management and queue. The resource manager of each cluster decides which query in the cluster’s queue should be executed next, based on logic described in the cluster’s [resource group configuration](https://prestodb.io/docs/current/admin/resource-groups.html). 
+
+This status quo of cluster-level queueing works fine to manage one or a few clusters, but quickly becomes problematic when managing a deployment of tens or hundreds of Presto clusters. Some issues include: 
+
+1. Due to the queue being fragmented across multiple clusters, and the admission decision being made at the cluster level, resource group admission decisions are not correct globally, although they could be correct locally. For example, it is possible that one cluster might admit a query of lower priority than a query in another cluster’s queue. 
+2. Since query concurrency control is applied on a per-cluster basis, global concurrency control can only be achieved, if the concurrency limit is a multiple of the number of clusters. For example, it is currently impossible to enforce a resource group with a concurrency limit of 1 query, across multiple clusters. 
+3. Load-balancing queries effectively across multiple clusters requires the load balancer to understand the resource group and queue state of each cluster. This is difficult to achieve, and in practice, stale information leads to poor load balancing. 
+4. Since the query is queued on the cluster itself, restarting clusters for maintenance reasons requires either that the queued queries be killed, or a lengthy drain time. 
+
+## Proposed Implementation
+
+We propose to add support in Presto for an optional queueing service that sits upstream of multiple Presto clusters. This queueing service will serve as the global entry-point for all queries from the client and will implement the QueuedStatement protocol. This queueing service is responsible for: 
+
+1. Authenticating the request 
+2. Adding the query to the global queue 
+3. Responding to clients polling for the queueing status of their query
+4. Picking the next query to execute 
+5. Selecting the best cluster for execution of this query
+6. Dispatching this query to the cluster
+7. Redirecting the client to the picked cluster
+
+The internal details of this queueing service are still under development, and this document focuses on the changes required to Presto for such a service to work. If there is interest from the community, we can provide an open-source reference implementation of this queueing service, once mature. 
+
+### Diagram
+![Presto Github Issue GRM](https://github.com/user-attachments/assets/7b2af8fd-5d82-411d-ac97-795e54e35164)
```

Comment:
You might want to add this diagram as a separate file like [here](https://github.com/prestodb/rfcs/pull/10/files#diff-7c4be749b64ffea6678fccf339b0f7430c613636a3ed4ca5a7a751abc67c71d9)